src/shared/components/person/profile.tsx

```diff
@@ -127,6 +130,11 @@ export class Profile extends Component<any, ProfileState> {
       this.state.personRes.person_view.person.id
     );
   }
+  get isBlocked() {
+    this.state.personBlocks = UserService.Instance.myUserInfo.person_blocks.
+    map(a => a.target.id).includes(this.state.personRes.person_view.person.id) 
+    return this.state.personBlocks
+  }
```

Comment:
You don't want to set that state here, do it below on line 631:

```ts
let blocks = updatePersonBlock(data);
this.state.personBlocks = blocks.map(a => a.target.id).includes(this.state.personRes.person_view.person.id);
this.setState(this.state);
```
Then you don't need an `isBlocked()` function, just `this.state.personBlocks` should work.

Also you should probably rename that to `personBlocked` to be more clear.